stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/listener/RegisterListener.java

```diff
@@ -96,12 +133,101 @@ public void notifyFailure(BError error) {
         }
     }
 
-    private void getMessages(Resource resource, Channel channel, String queue) {
+    /**
+     * Returns messages in String format.
+     *
+     * @param resource Ballerina resource function.
+     * @param channel  RabbitMQ channel.
+     * @param queue    Name of the queue.
+     */
+    private void getTextMessage(Resource resource, Channel channel, String queue) {
+        try {
+            DeliverCallback deliverCallback = (consumerTag, delivery) -> {
+                Executor.submit(resource, new ResponseCallback(), null, null,
+                        new BString(new String(delivery.getBody(), "UTF-8")));
+            };
+            channel.basicConsume(queue, true, deliverCallback, consumerTag -> {
+            });
+        } catch (IOException exception) {
+            throw new BallerinaException(exception);
+        }
+    }
+
+    /**
+     * Returns messages in int format.
+     *
+     * @param resource Ballerina resource function.
+     * @param channel  RabbitMQ channel.
+     * @param queue    Name of the queue.
+     */
+    private void getIntMessage(Resource resource, Channel channel, String queue) {
```

Comment:
Shall we rename to something like `dispatchIntMessage` since the method doesn't return anything? Shall we check all places?

Reply:
Fixed